### PR TITLE
Install as plugin

### LINKED ISSUE #11: inject does not work in dialogs
### Describe the bug

When accessing useFlag in dialogs (I am using quasar dialog), I get the following error:
Screenshot 2022-11-23 at 19 29 53

This happens because the values are provided from the root component.

The solution would be - not to use root level component for providing values, but install as a vue plugin. 

I can make a PR for the implementation if you want me to - just note that this would be a breaking change

### Steps to reproduce the bug

_No response_

### Expected behavior

_No response_

### Logs, error output, etc.

_No response_

### Screenshots

_No response_

### Additional context

_No response_

### Unleash version

_No response_

### Subscription type

_No response_

### Hosting type

_No response_

### SDK information (language and version)

_No response_

### PR DESCRIPTION
**Pull request by vojvodics**
## About the changes
Deleted the FlagProvider component and instead added option to install as a vue plugin. All the logic from the component is copied to the plugin. 
Things that changed:
- all provides are now scoped to the app using app.provide instead of provide imported from vue
- onMounted is deleted, and client starts on plugin install - not sure if that would be a problem

Closes #11 

### Important files

plugin.ts

## Discussion points

Is it a problem if client.start is called on plugin install?

### CODE REVIEW

**Comment by nunogois** [comment]
Hi @vojvodics, thank you for the PR!

I'm still unsure if we really need this, as mentioned in https://github.com/Unleash/proxy-client-vue/issues/11#issuecomment-1326457972, but assuming we want to add this, I think it would be important to **not** make it a breaking change, at least for now. I also think we should try to keep `proxy-client-vue` consistent with the other `proxy-client-*` SDK implementations if possible.

I would feel more comfortable with this if it's just an extra optional way of using the SDK, instead of breaking the existing one. If we go ahead with this, what do you think about keeping the original `FlagProvider` way (at least for now) and updating the README accordingly with both paths and examples?

**Comment by vojvodics** [comment]
@nunogois I'll update it so it won't be a breaking change.  Would you like me to refactor the code and reuse it in both the plugin and FlagProvider or are you OK with duplicated code?

**Comment by vojvodics** [comment]
I'll update README and examples as well

**Comment by nunogois** [comment]
> @nunogois I'll update it so it won't be a breaking change. Would you like me to refactor the code and reuse it in both the plugin and FlagProvider or are you OK with duplicated code?

If you think it's possible to reuse the same code for both that would be perfect, otherwise we can have some duplicate code for now.

> I'll update README and examples as well

That would be amazing 👍 

Thank you for understanding!

**Comment by vojvodics** [comment]
@nunogois I updated the PR. LMK if I need to fix anything else

**Comment by nunogois** [comment]
Thanks @vojvodics !

I checked out your fork to try it out but faced an issue building:
image
Seems related to: https://github.com/vuejs/core/issues/4294

However it worked if I used https://github.com/wheatjs/vite-plugin-vue-type-imports

But then when trying to use the plugin in my `main.ts` I get this type error:
image

How did you overcome this locally?

Other than that this PR seems super promising and I'm hoping we can get it merged soon!

**Comment by vojvodics** [comment]
@nunogois Added types and fix build. Can you check it out now?

File: README.md [hunk 1]
```diff
@@ -16,7 +16,27 @@ yarn add @unleash/proxy-client-vue
...
+const config = {
+  url: 'https://HOSTNAME/proxy',
+  clientKey: 'PROXYKEY',
+  refreshInterval: 15,
+  appName: 'your-app-name',
+  environment: 'dev'
+}
+
+const app = createApp(App)
+app.use(unleashPlugin, { config })
```
Comment (nunogois) [inline README.md:36]:
```suggestion
app.use(unleashPlugin, { config })
app.mount('#app')
```

File: README.md [hunk 2]
```diff
@@ -36,7 +56,29 @@ const config = {
...
+  clientKey: 'PROXYKEY',
+  refreshInterval: 15,
+  appName: 'your-app-name',
+  environment: 'dev'
+}
+
+const client = new UnleashClient(config)
+
+const app = createApp(App)
+app.use(unleashPlugin, { unleashClient: client })
```
Comment (nunogois) [inline README.md:79]:
```suggestion
app.use(unleashPlugin, { unleashClient: client })
app.mount('#app')
```

**Review by nunogois — APPROVED** [review]
LGTM - This is fantastic, great job! 

Only thing I noticed is that I had to upgrade Vue on my demo project for the `Plugin` types to match up.

Switching between the `plugin` and `FlagProvider` approach is super seamless and I'm glad we're not introducing a breaking change right away. We may phase out the `FlagProvider` approach eventually.

I noticed we were missing the "mount" lines in the README examples. WDYT about adding them?

**Comment by vojvodics** [comment]
@nunogois Thanks, updated. Sorry for the messy code and failing build - and thanks for noticing it and reviewing thoroughly!

I installed this on the reproduction example you provided and had no problem with the types. Which version of vue did you use? Is it a peer dependency issue?

**Comment by nunogois** [comment]
> @nunogois Thanks, updated. Sorry for the messy code and failing build - and thanks for noticing it and reviewing thoroughly!
> 
> I installed this on the reproduction example you provided and had no problem with the types. Which version of vue did you use? Is it a peer dependency issue?

No worries!

It might also have been because I was testing locally as a linked dependency.
Let's merge this for now and I'll do some more tests on my side.

Thank you for your contribution!

**Pull request merged**